lib/Models/WebProcessingServiceCatalogFunction.js

```diff
@@ -451,38 +442,69 @@ WebProcessingServiceCatalogFunction.prototype.invoke = function() {
 
     var that = this;
     when.all(createWpsDataInputsFromParameters(this)).then(function(dataInputs) {
-        dataInputs = dataInputs.join(";");
-
-        var uri = new URI(that.url).query({
-            service: 'WPS',
-            request: 'Execute',
-            version: '1.0.0',
-            Identifier: that.identifier,
-            DataInputs: dataInputs
-        });
-
-        if (that._statusSupported) {
-            uri.addQuery('status', true);
+        var proxyCacheDuration = '1d';
+        var promise;
+        if (that.useKvp) {
+            promise = loadResponseWithKvp(that, dataInputs, asyncResult, proxyCacheDuration);
+        } else {
+            promise = loadResponse(that, dataInputs, asyncResult, proxyCacheDuration);
         }
 
-        if (that._storeSupported) {
-            uri.addQuery('storeExecuteResponse', true);
-        }
-
-        var url = proxyCatalogItemUrl(that, uri.toString(), '1d');
-        var parameterValues = that.getParameterValues();
-
-        var promise = loadXML(url).then(function(xml) {
-            return handleExecuteResponse(that, parameterValues, asyncResult, xml);
-        });
-
         asyncResult.loadPromise = promise;
         asyncResult.isEnabled = true;
 
         return promise;
     });
 };
 
+function loadResponseWithKvp(that, dataInputs, asyncResult, proxyCacheDuration) {
+    dataInputs = dataInputs.join(";");
+    var uri = new URI(that.url).query({
+        service: 'WPS',
+        request: 'Execute',
+        version: '1.0.0',
+        Identifier: that.identifier,
+        DataInputs: dataInputs
+    });
+    if (that._statusSupported) {
+        uri.addQuery('status', true);
```

Comment:
It looks like `status` and `storeExecuteResponse` don't get added in the POST path.  Is that a problem?